store-pg/src/main/scala/com/socrata/pg/store/PGSecondary.scala

```diff
@@ -303,10 +380,14 @@ class PGSecondary(val config: Config) extends Secondary[SoQLType, SoQLValue] wit
 
       e match {
         case WorkingCopyCreated(copyInfo) =>
-          val theCopy = existingDataset.flatMap { dsInfo =>
-            val allCopies = pgu.datasetMapReader.allCopies(dsInfo)
-            // Either this copy or some newer copy
-            allCopies.find(existingCopyInfo => existingCopyInfo.copyNumber >= copyInfo.copyNumber)
+          val theCopy = existingDataset match {
+            case Some(dsInfo) =>
+              takeRollupLocks(pgu, dsInfo, upcomingRollups)
+              val allCopies = pgu.datasetMapReader.allCopies(dsInfo)
+              // Either this copy or some newer copy
+              allCopies.find(existingCopyInfo => existingCopyInfo.copyNumber >= copyInfo.copyNumber)
+            case None =>
+              None
```

Comment:
I rewrote it into a match because it felt weird having the takeRollupLocks side-effects inside flatMap.